tokenizers/src/utils/trie.rs

```diff
@@ -0,0 +1,301 @@
+use std::collections::HashMap;
+use std::hash::Hash;
+
+#[derive(Default)]
+pub struct TrieBuilder<Label> {
+    trie: Trie<Label>,
+}
+
+impl<Label: Eq + Hash + Copy> TrieBuilder<Label> {
+    pub fn push(&mut self, element: &[Label]) {
+        self.trie.push(element);
+    }
+
+    pub fn build(self) -> Trie<Label> {
+        self.trie
+    }
+}
+
+#[derive(Clone)]
+pub struct Trie<Label> {
```

Comment:
I know it's longer, but wouldn't `TokenTrie` be a better name?

Reply:
No, since it's not used only for token (it is used in Unigram too)

Reply:
Aren't unigrams still tokens ?

Reply:
Yes, but not for the Trie. Not in the way that would make `TokenTrie` any more understandable than `Trie` (The trie is only used to check common prefixes for instance, it doesn't "cut" tokens)